app/models/contract.rb

```diff
@@ -108,6 +110,10 @@ def self.by_plan_type(type)
 
   delegate :paid?, :to => :plan
 
+  def counter_culture_enabled?
+    !provider_account.scheduled_for_deletion?
```

Comment:
```suggestion
    !provider_account.scheduled_for_deletion? && !issuer.deleted?
```

Because imo it is the same if we are deleting the whole service but not the provider, then the plan is gonna be destroyed anyway so no need to updated the counter

Reply:
true true.